test/tools/libtesteth/TestHelper.cpp

```diff
@@ -198,13 +198,18 @@ void replaceLLLinState(json_spirit::mObject& _o)
 	}
 }
 
-std::vector<boost::filesystem::path> getJsonFiles(std::string const& _dirPath)
+std::vector<boost::filesystem::path> getJsonFiles(std::string const& _dirPath, std::string const& _pattern)
 {
 	vector<boost::filesystem::path> jsonFiles;
 	using Bdit = boost::filesystem::directory_iterator;
 	for (Bdit it(_dirPath); it != Bdit(); ++it)
+	{
 		if (boost::filesystem::is_regular_file(it->path()) && it->path().extension() == ".json")
-			jsonFiles.push_back(it->path());
+		{
+			if (_pattern.empty() || it->path().filename().string().find(_pattern.c_str()) != std::string::npos)
```

Comment:
If this function is supposed to return only one file when `_pattern` is non-empty, it's better in this case just to check if the file is there instead of iterating the directory.